### DISCUSSION TITLE
How to import plain maven projects

### DISCUSSION THREAD

**Discussion by phermsdorf** [eclipse-oomph/oomph#121] [Q&A] [answered]
Hi all,

i try to create a setup to import "plain" maven projects into an eclipse workspace with something like this:
```
  <setupTask
      xsi:type="git:GitCloneTask"
      id="MYID.git"
      location="${workspace.location/.git/}${@id.remoteURI|gitRepository}"
      remoteURI="https://bitbucket.org/mycompany/myapp.rcp.git"
      recursive="true">
    <description>my rcp</description>
  </setupTask>
  <setupTask
      xsi:type="projects:ProjectsImportTask">
    <sourceLocator
        rootFolder="${MYID.git.location}"
        locateNestedProjects="true"/>
  </setupTask>
```
but the setup log indicates that no projects could be found. Do i miss something here?

With plain maven projects i mean that there are .project and .classpath etc. files.
When importing them in the IDE these "project files" seem to be generated on the fly.

Is something similar possible with an Oomph setup?

Thanks for any hints on this.

Bye Peter

**Comment by merks** [comment]
Probably you need the project factory:
```
<?xml version="1.0" encoding="UTF-8"?>
<resources:SourceLocator
    xmi:version="2.0"
    xmlns:xmi="http://www.omg.org/XMI"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xmlns:resources="http://www.eclipse.org/oomph/resources/1.0"
    xsi:schemaLocation="http://www.eclipse.org/oomph/resources/1.0 https://raw.githubusercontent.com/eclipse-oomph/oomph/master/setups/models/Resources.ecore"
    rootFolder="${git.clone.blah.location}">
  <projectFactory
      xsi:type="resources:MavenProjectFactory"/>
</resources:SourceLocator>
```
The context menu on the source locator should give you that new child option.

**Reply by phermsdorf** [reply]
Thanks for your answer @merks.

I tried that but the installer still has the following problem:
```
Performing Projects Import
Importing projects from C:\workspace\main2\ws\.git\ 
```
At that location the root pom.xml is located and in subfolders the referenced modules.

When using the import Wizard (Maven-> Existing Maven Projects) the import succeeds.

Any idea what the problem could be?

**Reply by phermsdorf** [reply]
I've created a little reproducer:
[test.zip](https://github.com/user-attachments/files/18194343/test.zip)

Steps:

1. Start Oomph Installer in Advanced Mode
2. Choose "Eclipse IDE for Plugin and RAP Developers" as Product and Click Next
3. Add the test.setup as User Project and select it, Press Next and Continue with the Installation of the Setup
4. When the new Eclipse IDE Installation is launched the repo is cloned
5. the Project Import does not find any Projects to Import

Any idea what's wrong here?

Tank you, Bye Peter

**Reply by merks** [reply, marked as answer]
What I didn't notice from the XML but noticed looking at the example is that you aren't using a Maven Import Task but just a Project Import Task.  That's more obvious looking at the structured editor:

![image](https://github.com/user-attachments/assets/4aef1f57-7ea9-49f7-8d86-38c3a3a1c90c)

So while I didn't try it I thing this should work:
```
<?xml version="1.0" encoding="UTF-8"?>
<maven:MavenImportTask
    xmi:version="2.0"
    xmlns:xmi="http://www.omg.org/XMI"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xmlns:maven="http://www.eclipse.org/oomph/setup/maven/1.0"
    xmlns:resources="http://www.eclipse.org/oomph/resources/1.0"
    xsi:schemaLocation="http://www.eclipse.org/oomph/setup/maven/1.0 https://raw.githubusercontent.com/eclipse-oomph/oomph/master/setups/models/Maven.ecore http://www.eclipse.org/oomph/resources/1.0 https://raw.githubusercontent.com/eclipse-oomph/oomph/master/setups/models/Resources.ecore">
  <sourceLocator
      rootFolder="${MYID.git.location}"
      locateNestedProjects="true">
    <projectFactory
        xsi:type="resources:MavenProjectFactory"/>
  </sourceLocator>
</maven:MavenImportTask>
```
I'm not sure if you have the Oomph m2e extension installed.  The  context menu creation should help with that.  In any case, of course all the magic of synthesizing the .project files and such happens via the m2e infrastructure...

**Reply by phermsdorf** [reply]
Thanks @merks, that did the trick. I missed that there is a special Import Task for Maven Projects.
Now it's working.

Bonus Question: i see a setup Variable Target Platform where one can select an Eclipse release.
Is that some kind of default? Is it possible to get rid of this?

Background: we later set our own target platform with a target file ...

Thanks again.

Bye Peter

**Reply by merks** [reply]
You can set it to None:
```
<?xml version="1.0" encoding="UTF-8"?>
<setup:VariableTask
    xmi:version="2.0"
    xmlns:xmi="http://www.omg.org/XMI"
    xmlns:setup="http://www.eclipse.org/oomph/setup/1.0"
    name="eclipse.target.platform"
    value="None"
    storageURI="scope://Workspace"
    label="Target Platform">
  <annotation
      source="http://www.eclipse.org/oomph/setup/GlobalVariable"/>
  <description>This chooses the current target platform, reusing defintions from the org.eclipse.projects catalog, our logical parent, but we don't want to reuse that, so we set this to None.</description>
</setup:VariableTask>
```
